### PR TITLE
fix doc: Server-side data fetching with RLS

### PR DESCRIPTION
**Pull request by samuba**
took it from examples

### CODE REVIEW

File: packages/sveltekit/README.md [hunk 1]
```diff
@@ -222,37 +222,26 @@ For [row level security](https://supabase.com/docs/learn/auth-deep-dive/auth-row
 ```ts
 // src/routes/profile.ts
 import { supabaseServerClient, withApiAuth } from '@supabase/auth-helpers-sveltekit';
-import type { RequestHandler } from './__types/profile';
+import type { RequestHandler } from './__types/protected-route';
```
Comment (philip-weber) [inline packages/sveltekit/README.md:225]: Why did you change the type origin to `protected-route`? Aren't we on the file `profile.ts`?

File: packages/sveltekit/README.md [hunk 2]
```diff
@@ -222,37 +222,26 @@ For [row level security](https://supabase.com/docs/learn/auth-deep-dive/auth-row
...
 
 interface TestTable {
 	id: string;
 	created_at: string;
 }
 
 interface GetOutput {
-	user: User;
```
Comment (philip-weber) [inline packages/sveltekit/README.md:233]: To make this example consistent we still need the `user` in the body. Please have a look at line 213. The corresponding `profile.svelte` page expects a `user` prop.

File: packages/sveltekit/README.md [hunk 3]
```diff
@@ -222,37 +222,26 @@ For [row level security](https://supabase.com/docs/learn/auth-deep-dive/auth-row
 ```ts
 // src/routes/profile.ts
 import { supabaseServerClient, withApiAuth } from '@supabase/auth-helpers-sveltekit';
```
Comment (philip-weber) [inline packages/sveltekit/README.md:224]: You could add the `User` type to the imports to have it available in the `GetOutput` interface. Like `import { supabaseServerClient, withApiAuth, type User } from "@supabase/auth-helpers-sveltekit";`

We will need the `User` type further down.

File: packages/sveltekit/README.md [hunk 4]
```diff
@@ -222,37 +222,26 @@ For [row level security](https://supabase.com/docs/learn/auth-deep-dive/auth-row
...
-			user: locals.user
-		},
-		async () => {
-      const { data } = await supabaseServerClient(session.accessToken)
-        .from<TestTable>('test')
-        .select('*');
-
-			return {
-				body: {
-					user: locals.user,
```
Comment (philip-weber) [inline packages/sveltekit/README.md:250]: Like above. To make this example consistent we still need the user in the body. Please have a look at line 213. The corresponding profile.svelte page expects a user prop.

File: packages/sveltekit/README.md [hunk 5]
```diff
@@ -222,37 +222,26 @@ For [row level security](https://supabase.com/docs/learn/auth-deep-dive/auth-row
...
-				}
-			};
-		}
-	);
+export const get: RequestHandler<GetOutput> = async ({ locals, request }) =>
+	withApiAuth({ user: locals.user }, async () => {
+		const { data } = await supabaseServerClient(request).from<TestTable>('test').select('*');
+
+		return {
+			status: 200,
```
Comment (philip-weber) [inline packages/sveltekit/README.md:241]: Do we need to return the `200` status? I think this is returned implicitly if we do not add a different status code.

**Review by philip-weber — COMMENTED** [review]
Hi, I have added a couple of notes.

**Comment by david-plugge** [comment]
Should be solved
